### PR TITLE
implement config for width, focus behavior and filtering

### LINKED ISSUE #8: Width and optional filtering
First: thanks for making this! Much needed.

One question and one request:

a) how do I set the width/desired width of this box?

b) currently the box filters based on what is already in the field. This is not really what I want though, I'm basically using this as a search history thing where I keep the last 10 successful search terms in a VecDeque behind this as an iterable.
Also, I would like once you click the textedit that all the text gets highlighted, this is usually useful once you want to start a new search.

For b), can you give some guidance? (would also look into a PR if it is desired).

### PR DESCRIPTION
**Pull request by markusdd**
closes #8

I have added the features discussed in the ticket, was easier than I thought.
I would appreciate a merge and new release, I think we should not clutter the
ecosystem with a bunch of variants/forks.

This is all backwards compatible.
I kept the behavior at what it does currently unless the new functions get used.

### CODE REVIEW

File: examples/dropdown.rs [hunk 1]
```diff
@@ -11,12 +11,23 @@ impl App for ExampleApp {
...
-                    |ui, text| ui.selectable_label(false, text),
-                ));
+                ui.add(
+                    DropDownBox::from_iter(
+                        &self.items,
+                        "test_dropbox",
+                        &mut self.buf,
+                        |ui, text| ui.selectable_label(false, text),
+                    )
+                    // choose wether to filter the box items based on what is in the text edit already
```
Comment (ItsEthra) [inline examples/dropdown.rs:21]: wether -> whether (in other places as well)

File: examples/dropdown.rs [hunk 2]
```diff
@@ -11,12 +11,23 @@ impl App for ExampleApp {
...
+                    // default is true when this is not used
+                    .filter_by_input(false)
```
Comment (ItsEthra) [inline examples/dropdown.rs:23]: change `false` to `true` as it shows the default intended usage of this library

**Review by ItsEthra — CHANGES_REQUESTED** [review]

File: examples/dropdown.rs (hunk 1 above)
Comment (markusdd) [reply examples/dropdown.rs:21]: oooooops

File: examples/dropdown.rs (hunk 2 above)
Comment (markusdd) [reply examples/dropdown.rs:23]: ok

**Comment by markusdd** [comment]
requested changes done.

**Review by ItsEthra — APPROVED** [review]

**Comment by ItsEthra** [comment]
Thanks!

**Pull request merged**